### PR TITLE
pin setuptools to 64 for doc builds

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

**Comment by esc** [comment]
First seen as breakage on RTD builds:

```
building 'numba._dynfunc' extension
/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/dist.py:530: UserWarning: Normalizing '0.57.0dev0+220.g19a345749.dirty' to '0.57.0.dev0+220.g19a345749.dirty'
  warnings.warn(tmpl.format(**locals()))
/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/easy_install.py:147: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  EasyInstallDeprecationWarning,
Traceback (most recent call last):
  File "./setup.py", line 416, in <module>
    setup(**metadata)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 973, in run_commands
    self.run_command(cmd)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/install.py", line 74, in run
    self.do_egg_install()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/install.py", line 123, in do_egg_install
    self.run_command('bdist_egg')
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
    self.distribution.run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 165, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 151, in call_command
    self.run_command(cmdname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
    self.distribution.run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/command/install_lib.py", line 112, in build
    self.run_command('build_ext')
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
    self.distribution.run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "./setup.py", line 107, in run
    super().run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/command/build_ext.py", line 346, in run
    self.build_extensions()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/command/build_ext.py", line 466, in build_extensions
    self._build_extensions_serial()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/command/build_ext.py", line 492, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/setuptools/_distutils/command/build_ext.py", line 554, in build_extension
    depends=ext.depends,
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/ccompiler.py", line 89, in <lambda>
    m = lambda self, *args, **kw: func(self, *args, **kw)
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/ccompiler.py", line 273, in CCompiler_compile
    jobs = get_num_build_jobs()
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 90, in get_num_build_jobs
    from numpy.distutils.core import get_distribution
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/core.py", line 24, in <module>
    from numpy.distutils.command import config, config_compiler, \
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/command/config.py", line 19, in <module>
    from numpy.distutils.mingw32ccompiler import generate_manifest
  File "/home/docs/checkouts/readthedocs.org/user_builds/numba/conda/8350/lib/python3.7/site-packages/numpy/distutils/mingw32ccompiler.py", line 29, in <module>
    from distutils.msvccompiler import get_build_version as get_build_msvc_version
ModuleNotFoundError: No module named 'distutils.msvccompiler'
```

**Comment by esc** [comment]
Also, the error we are seeing on RTD was most likely caused by setuptools 65. Release notes include:

https://github.com/pypa/setuptools/blob/e6b911708e900f57c98e3b5b03af6de2953dbc76/CHANGES.rst?plain=1#L1-L15

File: docs/environment.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ dependencies:
   - llvmlite=0.40
   - numpy
   - numpydoc
-  - setuptools
+  - setuptools=64
```
Comment (stuartarchibald) [inline docs/environment.yml:12]:
```suggestion
  - setuptools<65
```
I'm not sure the availability of version 64 can be guaranteed, it's just anything that's less than 65 should be fine?

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
#8353 has the equivalent fix for gpuCI which passes - do you want to cherry-pick it into this?

File: docs/environment.yml (hunk 1 above)
Comment (esc) [reply docs/environment.yml]: fixed

**Comment by stuartarchibald** [comment]
Closing, merged as part of #8354.

**Pull request closed without merging**